core/chains/terra/terratxm/models.go

```diff
@@ -0,0 +1,78 @@
+package terratxm
+
+import (
+	"time"
+
+	"github.com/smartcontractkit/chainlink-terra/pkg/terra/client"
+
+	"github.com/smartcontractkit/terra.go/msg"
+)
+
+// State represents the state of a given terra msg
+// Happy path: Unstarted->Broadcasted->Confirmed
+type State string
+
+var (
+	// Unstarted means queued but not processed.
+	// Valid next states: Broadcasted, Errored (sim fails)
+	Unstarted State = "unstarted"
+	// Broadcasted means included in the mempool of a node.
+	// Valid next states: Confirmed (found onchain), Errored (tx expired waiting for confirmation)
+	Broadcasted State = "broadcasted"
+	// Confirmed means we're able to retrieve the txhash of the tx which broadcasted the msg.
+	// Valid next states: none, terminal state
+	Confirmed State = "confirmed"
+	// Errored means the msg reverted in simulation OR the tx containing the message timed out waiting to be confirmed
+	// TODO: when we add gas bumping, we'll address that timeout case
+	// Valid next states, none, terminal state
+	Errored State = "errored"
+)
+
+// TerraMsg a terra msg
+type TerraMsg struct {
+	ID         int64
+	ContractID string
+	State      State
+	Msg        []byte
+	TxHash     *string
+	CreatedAt  time.Time
+	UpdatedAt  time.Time
+
+	// In memory only
+	ExecuteContract *msg.ExecuteContract
```

Comment:
Is there a tag like `db:"-"` that can make this explicit?

Reply:
Addressed here https://github.com/smartcontractkit/chainlink-terra/pull/34/files#diff-be4a015634383dbff4d74701bff2696d640ad04d1ab083a82a9d92014ff8efc9R70